pkg/client/logs.go

```diff
@@ -228,8 +229,43 @@ type logStreamer struct {
 	client             kubernetes.Interface
 }
 
+func isContainerRunning(statuses *[]v1.ContainerStatus, containerName string) bool {
+	for _, cs := range *statuses {
+
+		fmt.Printf("container %v, state= %v\n", containerName, cs.State)
+		if cs.Name == containerName && cs.State.Running != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func (l *logStreamer) waitForContainerRunning() error {
+	backoffSeconds := 1 * time.Second
+	for {
+		pod, err := l.client.CoreV1().Pods(l.ns).Get(l.pod, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get pod [%s/%s]", l.ns, l.pod)
+		}
+
+		if isContainerRunning(&pod.Status.ContainerStatuses, l.container) {
+			return nil
+		}
+
+		time.Sleep(backoffSeconds)
+		backoffSeconds *= 2
```

Comment:
So we're doing this backoff with a 2N wait time but if the pod fails we will just always be waiting.

I suggested removing the fmt.Print line from the `isContainerRunning` check, but it seems reasonable to me to put one here so people know what is happening:

```
fmt.Printf("container %v, state= %v will attempt to stream logs in %v seconds\n", containerName, cs.State, backoffSeconds)
```